aspnetcore/migration/1x-to-2x/identity-2x.md

```diff
@@ -0,0 +1,325 @@
+
+    // If you want to tweak Identity cookies, they're no longer part of IdentityOptions.
+    services.ConfigureApplicationCookie(options => options.LoginPath = "/Account/LogIn");
+    services.AddAuthentication()
+            .AddFacebook(options => {
+                options.AppId = Configuration["facebook:appid"];
+                options.AppSecret = Configuration["facebook:appsecret"];
+            });
+}
+
+public void Configure(IApplicationBuilder app, ILoggerFactory loggerfactory) {
+    app.UseAuthentication();
+}
+```
+
+The `UseAuthentication` method adds a single authentication middleware component which is responsible for automatic authentication and the handling of remote authentication requests. It replaces all of the individual middleware components with a single, common middleware component.
+
+Below are 2.0 migration instructions for each major authentication scheme.
+
+### Cookie-based Authentication
+Select one of the two options below, and make the necessary changes in *Startup.cs*:
+
+1. Use cookies with Identity
+    - Invoke the `UseAuthentication` method in the `Configure` method:
+
+        ```csharp
+        app.UseAuthentication();
+        ```
+
+    - Invoke the `AddIdentity` method in the `ConfigureServices` method to add the cookie authentication services.
+    - Optionally, invoke the `ConfigureApplicationCookie` or `ConfigureExternalCookie` method in the `ConfigureServices` method to tweak the Identity cookie settings.
+
+        ```csharp
+        services.AddIdentity<ApplicationUser, IdentityRole>()
+                .AddEntityFrameworkStores<ApplicationDbContext>()
+                .AddDefaultTokenProviders();
+    
+        services.ConfigureApplicationCookie(options => options.LoginPath = "/Account/LogIn");
+        ```
+
+2. Use cookies without Identity
+    - Replace the `UseCookieAuthentication` method call in the `Configure` method with `UseAuthentication`:
+  
+        ```csharp
+        app.UseAuthentication();
+        ```
+ 
+    - Invoke the `AddAuthentication` and `AddCookie` methods in the `ConfigureServices` method:
+
+        ```csharp
+        // If you don't want the cookie to be automatically authenticated, remove the 
+        // CookieAuthenticationDefaults.AuthenticationScheme parameter passed AddAuthentication
+        services.AddAuthentication(CookieAuthenticationDefaults.AuthenticationScheme)
+                .AddCookie(options => {
+                    options.LoginPath = "/Account/LogIn";
+                    options.LogoutPath = "/Account/LogOff";
+                });
+        ```
+
+### JWT Bearer Authentication
+Make the following changes in *Startup.cs*:
+- Replace the `UseJwtBearerAuthentication` method call in the `Configure` method with `UseAuthentication`:
+ 
+    ```csharp
+    app.UseAuthentication();
+    ```
+
+- Invoke the `AddJwtBearer` method in the `ConfigureServices` method:
+
+    ```csharp
+    services.AddAuthentication().AddJwtBearer(options => {
```

Comment:
Since JwtBearer is used without Identity the default scheme should be set.
`services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme).AddJwtBearer(options =>`